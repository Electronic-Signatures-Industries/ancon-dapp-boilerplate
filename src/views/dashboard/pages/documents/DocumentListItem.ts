interface Item {
  contentType: string;
}

export interface DocumentListItem {
  folder: Item | Item[];
}
