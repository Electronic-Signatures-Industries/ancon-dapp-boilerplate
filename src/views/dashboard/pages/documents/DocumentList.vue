<template>
  <v-card>
    <v-card-title>Documentos</v-card-title>

    <v-card-text v-if="items.length < 1">
      No hay documentos disponibles.
    </v-card-text>

    <v-card-text v-else>
      <v-treeview
        :active="selectedValue"
        @update:active="updateSelected"
        :items="items"
        activatable
      >
        <template v-slot:prepend="{ item, open }">
          <v-icon v-if="!item.folder.contentType">
            {{ open ? "mdi-folder-open" : "mdi-folder" }}
          </v-icon>
          <v-icon v-else>
            {{ getFileIcon(item.folder.contentType) }}
          </v-icon>
          {{ item.folder.length }} archivo(s)
        </template>
      </v-treeview>
    </v-card-text>

    <v-card-actions>
      <v-spacer />

      <slot name="buttons"/>
    </v-card-actions>
  </v-card>
</template>

<script lang="ts">
import Vue from 'vue';
import { Component, Emit, Prop } from 'vue-property-decorator';
import { FileIcons } from './FileIcons';
import { DocumentListItem } from './DocumentListItem';

@Component({})
export default class DocumentList extends Vue {
  @Prop()
  readonly selectedValue: any;

  @Prop()
  readonly items: DocumentListItem[];

  @Emit('update:selectedValue')
  updateSelected(newValue) {
    return newValue;
  }

  getFileIcon(contentType: string): string {
    return FileIcons[contentType]
  }
}
</script>