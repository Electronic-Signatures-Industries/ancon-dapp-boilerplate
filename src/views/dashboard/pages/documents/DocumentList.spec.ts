import { createLocalVue, shallowMount } from "@vue/test-utils";
import Vuetify from "vuetify";
import DocumentList from "./DocumentList.vue";
import { DocumentListItem } from "./DocumentListItem";

const listItems: DocumentListItem[] = [
  {
    folder: [
      {
        contentType: "text/html",
      },
    ],
  },
  {
    folder: {
      contentType: "application/json",
    },
  },
];

describe("DocumentList.vue", () => {
  it("should match snapshot", () => {
    const component = shallowMount(DocumentList, {
      localVue: createLocalVue(),
      vuetify: new Vuetify(),
      propsData: {
        items: listItems,
      },
    });
    expect(component).toMatchSnapshot();
  });
});
