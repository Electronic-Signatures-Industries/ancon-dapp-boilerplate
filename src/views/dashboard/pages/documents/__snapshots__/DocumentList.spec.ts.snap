// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentList.vue should match snapshot 1`] = `
<v-card-stub loaderheight="4" tag="div">
  <v-card-title-stub>Documentos</v-card-title-stub>
  <v-card-text-stub>
    <v-treeview-stub active="" items="[object Object],[object Object]" open="" value="" activatable="true" activeclass="v-treeview-node--active" color="primary" expandicon="$subgroup" indeterminateicon="$checkboxIndeterminate" itemchildren="children" itemdisabled="disabled" itemkey="id" itemtext="name" loadingicon="$loading" officon="$checkboxOff" onicon="$checkboxOn" selectedcolor="accent" selectiontype="leaf"></v-treeview-stub>
  </v-card-text-stub>
  <v-card-actions-stub>
    <v-spacer-stub></v-spacer-stub>
  </v-card-actions-stub>
</v-card-stub>
`;
